src/main/java/Storage.java

```diff
@@ -45,15 +45,8 @@ else if(str.charAt(1) == 'D') {
                 String taskName = str.substring(7, openBracket - 1);
                 deadline input = new deadline(taskName);
 
-                SimpleDateFormat to = new SimpleDateFormat("yyyy-MM-dd");
-                SimpleDateFormat from = new SimpleDateFormat("MMM dd yyyy");
-                try {
-                    Date date = from.parse(str.substring(openBracket + 5, closeBracket));
-                    String result = to.format(date);
-                    input.setDate(new formatDate(result));
-                } catch (ParseException e) {
-                    System.out.println("Invalid Date Format!");
-                }
+                String result = input.correctDateFormat(str.substring(openBracket + 5, closeBracket));
+                input.setDate(new formatDate(result));
 
                 if (str.charAt(4) == 'X') {
```

Comment:
Maybe I could ask why these checks are repeated / done in storage repeatedly? The way I might do it would be to create a static factory class within each of deadline/todo/event, and call an appropriate constructor depending on whether it is 'E', 'T' or 'D'. 

In general, I think you could consider trying to abstract out the parsing of the saveData and creation into a seperate parser class / within each Event/Deadline/Task class itself, and not attempt to create it within Storage :D Could save on repetition of code, and it might make use of better OOP principles!